addons/hw_drivers/tools/helpers.py

```diff
@@ -198,15 +199,17 @@ def generate_password():
     """
     Generate an unique code to secure raspberry pi
     """
-    alphabet = 'abcdefghijkmnpqrstuvwxyz23456789'
-    password = ''.join(secrets.choice(alphabet) for i in range(12))
-    shadow_password = crypt.crypt(password, crypt.mksalt())
-    subprocess.call(('sudo', 'usermod', '-p', shadow_password, 'pi'))
+    if main.ssh_service:
+        alphabet = 'abcdefghijkmnpqrstuvwxyz23456789'
+        password = ''.join(secrets.choice(alphabet) for i in range(12))
+        shadow_password = crypt.crypt(password, crypt.mksalt())
+        subprocess.call(('sudo', 'usermod', '-p', shadow_password, 'pi'))
 
-    with writable():
-        subprocess.call(('sudo', 'cp', '/etc/shadow', '/root_bypass_ramdisks/etc/shadow'))
+        with writable():
+            subprocess.call(('sudo', 'cp', '/etc/shadow', '/root_bypass_ramdisks/etc/shadow'))
 
-    return password
+        return password
+    return "ssh_service not activate"
```

Comment:
Return a better description ("Activate ssh service in the db"?)

Reply:
Enable ssh in the db first

Reply:
maybe better "Enable ssh first in the db"